pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -682,7 +682,7 @@ public void reloadLookUp() throws PulsarClientException {
             Optional<ConsumerBase<?>> subscriber = consumers.keySet().stream()
                     .filter(c -> c.getSubType().equals(PulsarApi.CommandSubscribe.SubType.Shared))
                     .filter(c -> conf.getTopicNames().contains(c.getTopic()))
-                    .filter(c -> c.getSubscription().equals(conf.getSubscriptionName()))
+                    .filter(c -> c.getConsumerName().equals(conf.getConsumerName()))
```

Comment:
I think it needs both subscription name and consumer name. 

Subscription name is the "consumer group" name. Consumer name is for identifying a consumer in the group.

Reply:
Sorry, I missed this in the hurry.
Now is fixed proper test to assert this. :+1: